src/Microsoft.AspNetCore.SignalR.Client/HttpConnectionFactory.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Http.Connections.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public class HttpConnectionFactory : IConnectionFactory
+    {
+        private readonly HttpConnectionOptions _options;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public HttpConnectionFactory(IOptions<HttpConnectionOptions> options, ILoggerFactory loggerFactory)
+        {
+            _options = options.Value;
```

Comment:
Check arguments for null.

Reply:
Just FYI, in practice these can't be null because this is activated by the DI container. It is a public class though so I'll do it anyways although I *despise* the gratuitous layers of argument checking in .NET.